miio/device.py

```diff
@@ -155,6 +161,32 @@ def _fetch_info(self) -> DeviceInfo:
                 "Unable to request miIO.info from the device"
             ) from ex
 
+    def _initialize_descriptors(self) -> None:
+        """Cache all the descriptors once on the first call."""
+        if self._sensors is not None:
+            return
+
+        status = self.status()
+
+        # Sensors
+        self._sensors = status.sensors()
+
+        # Settings
+        self._settings = status.settings()
+        for setting in self._settings.values():
+            if setting.setter_name is not None:
+                setting.setter = getattr(self, setting.setter_name)
+            if setting.setter is None:
+                raise Exception(
+                    f"Neither setter or setter_name was defined for {setting}"
+                )
+            if (
+                setting.type == SettingType.Enum
+                and setting.choices_attribute is not None
```

Comment:
Hmmh, I suppose we don't currently have tests for this case, but wouldn't we just fail silently if only `choices` if defined?

Reply:
Well ether `choices` or `choices_attribute` needs to be defined in order to get the `SettingType.Enum` in the decorator. We only need `choices` so if `choices` is defined (as a static enum) everything will work just fine.
If `choices_attribute` is defined (for dynamic enums) the `choices` will be set from a call to the `choices_attribute` function.

So no we will not fail silently